cpp/src/binaryop/compiled/binops_custom.cuh

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "binary_ops.hpp"
+#include "operation.cuh"
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/exec_policy.hpp>
+
+namespace cudf {
+
+namespace binops {
+namespace compiled {
+
+namespace {
```

Comment:
Don't use anonymous namespaces in headers.

Reply:
I didn't want these structs to be in symbol table and used only within a TU.
This header is used in all `.cu` files for each operator.

Reply:
Hmm. This is something I've been doing as well. @jrhemstad, wouldn't the alternative imply exposing functions that would've otherwise been concealed in the TUs in which they are called?
Is it preferable to expose those symbols?

Reply:
https://github.com/rapidsai/cudf/blob/branch-21.06/cpp/docs/DEVELOPER_GUIDE.md#anonymous-namespace

https://wiki.sei.cmu.edu/confluence/display/cplusplus/DCL59-CPP.+Do+not+define+an+unnamed+namespace+in+a+header+file

Reply:
removed anonymous namespace.